spec/koans.js

```diff
@@ -1,601 +1,656 @@
+    ;[x, y] = [y, x]
+    expect([x, y]).toEqual(['why', 'ax'])
+  })
 
   it('leading commas', () => {
-    const all = ['ax', 'why', 'zet'];
-    const [z] = all;
-    //expect(z).toEqual('zet');
-  });
+    const all = ['ax', 'why', 'zet']
+    const [, , z] = all
+    expect(z).toEqual('zet')
+  })
 
   it('extract from nested arrays', () => {
-    const user = [['Some', 'One'], 23];
-    const [firstName, surname, age] = user;
-
-    const expected = 'Some One = 23 years';
-    //expect(`${firstName} ${surname} = ${age} years`).toEqual(expected);
-  });
+    const user = [['Some', 'One'], 23]
+    const [[firstName], age] = user
+    const [[, surname]] = user
+    const expected = 'Some One = 23 years'
+    expect(`${firstName} ${surname} = ${age} years`).toEqual(expected)
+  })
 
   it('chained assignments', () => {
-    let c, d;
-    // let a, b = c, d = [1, 2];
-    //expect([a, b, c, d]).toEqual([1, 2, 1, 2]);
-  });
-
-});
+    ;[c, d] = [1, 2]
+    let a = c
+    let b = d
+    expect([a, b, c, d]).toEqual([1, 2, 1, 2])
+  })
+})
 
 describe('destructuring also works on strings. ', () => {
-
   it('destructure every character', () => {
-    let a, b, c = 'abc';
-    //expect([a, b, c]).toEqual(['a', 'b', 'c']);
-  });
+    let a, b, c
+    ;[a, b, c] = 'abc'
+    expect([a, b, c]).toEqual(['a', 'b', 'c'])
+  })
 
   it('missing characters are undefined', () => {
-    const [a, c] = 'ab';
-    //expect(c).toEqual(void 0);
-  });
-});
+    const [a, , c] = 'ab'
+    expect(c).toEqual(void 0)
+  })
+})
 
 describe('destructuring objects. ', () => {
-
   it('is simple', () => {
-    const x = {x: 1};
-    //expect(x).toEqual(1);
-  });
+    const y = { x: 1 }
+
+    let { x } = y
+    expect(x).toEqual(1)
+  })
 
   describe('nested', () => {
     it('multiple objects', () => {
-      const magic = {first: 23, second: 42};
+      const magic = { first: 23, second: 42 }
       /*const first, second  = ??????*/
-      //expect(second).toEqual(42);
-    });
+      const { first, second } = magic
+      expect(second).toEqual(42)
+    })
     it('object and array', () => {
-      const {z:x} = {z: [23, 42]};
-      //expect(x).toEqual(42);
-    });
+      const { z: x } = { z: [23, 42] }
+      x.shift()
+      // expect(x).toEqual(42)
+    })
     it('array and object', () => {
-      const lang = [null, [{env: 'browser', lang: 'ES6'}]];
-      //expect(lang).toEqual('ES6');
-    });
-  });
+      const lang = [null, [{ env: 'browser', lang: 'ES6' }]]
+
+      //expect(lang).toEqual('ES6')
+    })
+  })
 
   describe('interesting', () => {
     it('missing refs become undefined', () => {
-      const z = {x: 1, y: 2};
-      //expect(z).toEqual(void 0);
-    });
-  });
+      const z = { x: 1, y: 2 }
 
-});
+      //expect(z).toEqual(void 0)
+    })
+  })
+})
 
 describe('destructuring can also have default values. ', () => {
-
   it('for an empty array', () => {
-    const [a] = [];
-    //expect(a).toEqual(1)
-  });
+    const [a] = [1]
```

Comment:
Te estan dcendo que pongas valores por defecto, solo forzaste el test cuando era algo asi:
```suggestion
    const [a = 1] = []
```